include/javabind/javabind.hpp

```diff
@@ -181,7 +182,80 @@ static jint java_initialization_impl(JavaVM* vm, void (*initializer)())
+                return JNI_ERR;
+            }
+
+            std::string values_sig = std::string("()[L").append(cn).append(";");
+            jmethodID values_ref = env->GetStaticMethodID(cls.ref(), "values", values_sig.data());
+            if (values_ref == nullptr) {
+                javabind::throw_exception(env,
+                    msg() << "Cannot find static method 'values' with signature '" << values_sig << "' in registered enum class '" << class_name
+                );
+                return JNI_ERR;
+            }
+
+            jmethodID ordinal_ref = env->GetMethodID(cls.ref(), "ordinal", "()I");
+            if (ordinal_ref == nullptr) {
+                javabind::throw_exception(env,
+                    msg() << "Cannot find method 'ordinal' with signature '()I' in enum value of class '" << class_name << "'"
+                );
+                return JNI_ERR;
+            }
+
+            jmethodID name_ref = env->GetMethodID(cls.ref(), "name", "()Ljava/lang/String;");
+            if (name_ref == nullptr) {
+                javabind::throw_exception(env,
+                    msg() << "Cannot find method 'name' with signature '()Ljava/lang/String;' in enum value of class '" << class_name << "'"
+                );
+                return JNI_ERR;
+            }
+
+            jarray values_arr = static_cast<jarray>(env->CallStaticObjectMethod(cls.ref(), values_ref));
+            if (values_arr == nullptr) {
+                javabind::throw_exception(env,
+                    msg() << "Static method 'values' with signature '" << values_sig << "' in registered enum class '" << class_name << "' returned null"
+                );
+                return JNI_ERR;
+            }
+
+            std::unordered_map<std::string, EnumValue> values;
+
+            for (jsize i = 0; i < env->GetArrayLength(values_arr); ++i) {
+                jobject value = env->GetObjectArrayElement(static_cast<jobjectArray>(values_arr), i);
+                if (value == nullptr) {
+                    javabind::throw_exception(env,
+                        msg() << "Element " << i << " of static method 'values' in enum class '" << class_name << "' returned null"
+                    );
+                    return JNI_ERR;
+                }
+
+                std::string name = arg_type_t<std::string>::native_value(env, static_cast<jstring>(env->CallObjectMethod(value, name_ref)));
+
+                if (!bindings.contains(name)) {
+                    javabind::throw_exception(env,
+                        msg() << "Enum value '" << name << "' in class '" << class_name << "' is not registered in C++ code"
+                    );
+                    return JNI_ERR;
+                }
+
+                jint ordinal = env->CallIntMethod(value, ordinal_ref);
+                values.emplace(name, EnumValue { env->NewGlobalRef(value), ordinal });
+            }
+
+            bindings.initialize(values);
+        }
+    } catch (std::exception& e) {
```

Comment:
I have recently added a GitHub Actions workflow to validate the code on both Linux and Windows. It turns out MSVC emits a warning/error for unreferenced local variables like `e` in this context. I had to edit the code (with multiple rounds of pushes to `master`) to remove these orphan variables. (Unfortunately, GitHub didn't show the workflow unless it was pushed to `master`.)